src/Disks/IO/ReadBufferFromWebServer.h

```diff
@@ -33,7 +34,7 @@ class ReadBufferFromWebServer : public ReadBufferFromFileBase
 
     void setReadUntilPosition(size_t position) override;
 
-    size_t getFileOffsetOfBufferEnd() const override { return offset; }
+    size_t getFileOffsetOfBufferEnd() const override { return offset.load(std::memory_order_relaxed); }
```

Comment:
I don't entirely understand these changes - why is the offset concurrently modified and read?

Reply:
It happens because cached buffer does some logging in SCOPE_EXIT when it already gave up ownership of that buffer (I think it is ok as it is only logging). (Cached buffers can pass ownership of impl (s3 buffer, for example) to each other).

Reply:
Ok, copy-paste it to the code, and it will LGTM.

Reply:
Apparently, @azat can do it.

Reply:
Actually I remember that this was not the first time, so maybe it should be refactored. For now - added comments.